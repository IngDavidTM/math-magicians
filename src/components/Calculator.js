import './Calculator.css';
import React from 'react';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="calculator">
        <p className="screen">0</p>
        <div className="keys">
          <button type="button">AC</button>
          <button type="button">+/-</button>
          <button type="button">%</button>
          <button type="button" className="orange">รท</button>
          <button type="button">7</button>
          <button type="button">8</button>
          <button type="button">9</button>
          <button type="button" className="orange">x</button>
          <button type="button">4</button>
          <button type="button">5</button>
          <button type="button">6</button>
          <button type="button" className="orange">-</button>
          <button type="button">1</button>
          <button type="button">2</button>
          <button type="button">3</button>
          <button type="button" className="orange">+</button>
          <button type="button" className="btn0">0</button>
          <button type="button">.</button>
          <button type="button" className="orange">=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
