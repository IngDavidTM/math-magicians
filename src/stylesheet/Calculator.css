.calculator {
  background-color: #848494;
  display: flex;
  flex-direction: column;
  max-width: 500px;
  width: 100%;
}

.screen {
  color: #fff;
  font-size: 1.3em;
  margin: 20px 5px;
  text-align: end;
}

.keys {
  display: flex;
  flex-wrap: wrap;
}

button {
  min-width: 25%;
  border: none;
  background-color: #e3e3e3;
  border-top: 1px solid #bbbcbc;
  border-left: 1px solid #bbbcbc;
  cursor: pointer;
  color: #000;
  padding: 20px 0;
}

.btn0 {
  min-width: 50%;
}

.orange {
  background-color: #f3933c;
}

button:hover {
  background-color: #e3e3e3c0;
}

.orange:hover {
  background-color: #f3943cd0;
}
