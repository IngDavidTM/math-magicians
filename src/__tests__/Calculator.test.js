import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('should render Calculator', () => {
  const renderCalculator = renderer.create(<Calculator />).toJSON();
  expect(renderCalculator).toMatchSnapshot();
});
