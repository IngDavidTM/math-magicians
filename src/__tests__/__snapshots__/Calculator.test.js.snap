// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Calculator 1`] = `
<div
  className="App"
>
  <div
    className="calculator"
  >
    <p
      className="screen"
    >
      0
    </p>
    <div
      className="keys"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
